timApp/velp/velpgroups.py

```diff
@@ -51,6 +49,56 @@ class CreatedVelpGroup:
     created_new_group: bool = True
 
 
+@dataclass
+class GroupSelection:
+    """Stores information on a velp group's selection status"""
+
+    id: int
+    selected: bool
+
+    def to_json(self) -> dict[str, int | bool]:
+        return {"id": self.id, "selected": self.selected}
+
+
+@dataclass
+class VelpGroupSelectionInfo:
+    """Stores information on selected and default velp groups for a document. Used for
+    passing velp group selection data to the user interface.
+    """
+
+    target_ids: list[str]
+    selections: list[list[GroupSelection]]
+
+    def append(self, target_id: str, gs: GroupSelection) -> None:
+        if len(target_id) == 0:
+            return
+        else:
+            index = -1
+            for i in range(len(self.target_ids)):
+                if target_id == self.target_ids[i]:
+                    index = i
+                    break
```

Comment:
Pythonissa sekä indeksin että arvon voi lukea samanaikaisesti `enumerate`-metodilla. Esimerkiksi:

```py
            for i, tid in enumerate(self.target_ids):
                if target_id == tid:
                    index = i
                    break
```

Mutta melestäni tämän saa vähän paremmin luettavaksi käyttämällä `next`-metodia sekä iteraattoria:

```py
index = next((i for i, tid in enumerate(self.target_ids) if target_id == tid), -1)
```

Tässä siis `(i for i, tid in enumerate(self.target_ids) if target_id == tid)` on sama kuin
```py
for i, tid in enumerate(self.target_ids)
  if target_id == tid
    yield i
```

(`yield i` tarkoittaa suunnilleen "palauta muuttujan `i` arvo ja pysäytä iterointi").  Puolestaan `next()` on funktio, joka ottaa iteraattorin ja palauttaa ensimmäisen arvon, joka sille `yield`lla annetaan. Jos arvoa ei ole, palautetaan `-1` (toinen parametri on oletusarvo).

Reply:
Joo, tuo on paljon nätimpi tapa tehdä tuo. :)